### PR TITLE
bugfix: driver: esp32: eth_mdio: fix clock and function calls

### LINKED ISSUE #74440: ESP32 ethernet interface do not start
**Describe the bug**
As discussed in #68907, ESP32 ethernet interface won't be up.

**To Reproduce**
1. mkdir build; cd build
2. west build -p -b esp32_ethernet_kit/esp32/procpu samples/boards/esp32/ethernet/

**Expected behavior**
Interface up and running + IP addresses:
```c
*** Booting Zephyr OS build v3.7.0-rc1-19-g02ad7ef1f9cd ***
[00:00:01.114,000] <inf> net_config: Initializing network
[00:00:01.114,000] <inf> net_config: Waiting interface 1 (0x3ffb5f08) to be up...
[00:00:04.708,000] <inf> phy_mii: PHY (1) Link speed 100 Mb, full duplex

[00:00:04.708,000] <inf> net_config: Interface 1 (0x3ffb5f08) coming up
[00:00:04.708,000] <inf> net_config: Running dhcpv4 client...
[00:00:20.750,000] <inf> net_dhcpv4: Received: 192.168.68.124
[00:00:20.751,000] <inf> net_config: IPv4 address: 192.168.68.124
[00:00:20.751,000] <inf> net_config: Lease time: 7200 seconds
[00:00:20.751,000] <inf> net_config: Subnet: 255.255.255.0
[00:00:20.751,000] <inf> net_config: Router: 192.168.68.1
```

**Impact**
Showstopper as ethernet won´t  work.
```c
*** Booting Zephyr OS build v3.7.0-rc1-19-g02ad7ef1f9cd ***
[00:00:01.114,000] <inf> net_config: Initializing network
[00:00:01.114,000] <inf> net_config: Waiting interface 1 (0x3ffb5f08) to be up...
[00:00:04.708,000] <inf> phy_mii: PHY (1) Link speed 100 Mb, full duplex
...
...
```

### LINKED ISSUE #74632: esp32: can´t built project using base64 sources
**Describe the bug**
Whenever `CONFIG_BASE64=y` is used with Wi-Fi, build will fail due to base64 sources included in `wpa_supplicant`.

Please also mention any information which could help others to understand
the problem you're facing:
 - Any ESP32x board

**To Reproduce**
1. Update `sample/net/wifi_test.c`:
```c
/*
 * Copyright (c) 2016 Intel Corporation.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

#include <zephyr/kernel.h>
#include <errno.h>
#include <zephyr/sys/base64.h>

int main(void)
{
    int ret;
    char *input = "Hello, World!";
    char output[64];
    size_t olen;

    ret = base64_encode(output, sizeof(output), &olen, input, strlen(input));
    if (ret == 0) {
        output[olen] = '\0';
        printk("Encoded: %s\n", output);
    }

    return 0;
}
```
 
2. rm -rf build; west build -b esp32c3_devkitm samples/net/wifi/ -DCONFIG_BASE64=y
3. See error:
```c
multiple definition of `base64_decode'; zephyr/libzephyr.a(base64.c.obj):/home/sylvio/espressif/zephyr/zephyr/lib/utils/base64.c:126: first defined here
collect2: error: ld returned 1 exit status
```

**Expected behavior**
Code working.

### PR DESCRIPTION
**Pull request by sylvioalves**
Both MDIO and Ethernet drivers share the same clock subsystem. After clock control update in #73807, clock_control_on() now returns -EALREADY for already initialized clock subsystem. As a result, ethernet driver won't initialize as needed. 
Due to the same clock control and hal updates, internal functions required updates.

Fixes #74440
Fixes #74632

As this PR updates hal_espressif, it also fixes #74632.

### CODE REVIEW

**Review by LucasTambor — DISMISSED** [review]

**Comment by sylvioalves** [comment]
It is also important to mention that current `clock_control_api` test interface considers that `-EALREADY` is an error: 
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/clock_control/clock_control_api/src/test_clock_control.c#L261

**Comment by sylvioalves** [comment]
Bebase again to check whether Babblesim stops failing.

File: drivers/ethernet/eth_esp32.c [hunk 1]
```diff
@@ -255,7 +287,9 @@ int eth_esp32_initialize(const struct device *dev)
 
 		emac_hal_iomux_rmii_clk_output(ref_clk_gpio);
 		emac_ll_clock_enable_rmii_output(dev_data->hal.ext_regs);
-		rtc_clk_apll_enable(true, 0, 0, 6, 2);
+		periph_rtc_apll_acquire();
+		emac_config_apll_clock();
```
Comment (bbilas) [inline drivers/ethernet/eth_esp32.c:291]: Why don't you handle the return code?

File: drivers/ethernet/eth_esp32.c [hunk 2]
```diff
@@ -200,6 +202,35 @@ static void phy_link_state_changed(const struct device *phy_dev,
...
+	/* If the difference of real APLL frequency
+	 * is not within 50 ppm, i.e. 2500 Hz,
+	 * the APLL is unavailable
+	 */
+	if (abs((int)real_freq - (int)expt_freq) > 2500) {
+		LOG_ERR("The APLL is working at an unusable frequency");
+		return -EIO;
+	}
+
+	return ESP_OK;
```
Comment (bbilas) [inline drivers/ethernet/eth_esp32.c:230]:
```suggestion
	return 0;
```

File: drivers/ethernet/eth_esp32.c [hunk 3]
```diff
@@ -200,6 +202,35 @@ static void phy_link_state_changed(const struct device *phy_dev,
...
+	 * the APLL is unavailable
+	 */
+	if (abs((int)real_freq - (int)expt_freq) > 2500) {
+		LOG_ERR("The APLL is working at an unusable frequency");
+		return -EIO;
+	}
+
+	return 0;
+}
+#endif /* DT_INST_NODE_HAS_PROP(0, bt_reg_on_gpios) */
```
Comment (bbilas) [inline drivers/ethernet/eth_esp32.c:232]:
```suggestion
#endif /* DT_INST_NODE_HAS_PROP(0, ref_clk_output_gpios) */
```

**Review by bbilas — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Pull request merged**